llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -444,10 +444,35 @@ class WAWEdges : public ScheduleDAGMutation {
     auto *RI = static_cast<const AIEBaseRegisterInfo *>(TRI);
     LivePhysRegs LiveRegs;
     LiveRegs.init(*TRI);
-    // Reserved registers are considered always live
-    for (MCPhysReg PhysReg : MRI.getReservedRegs().set_bits()) {
-      if (RI->isSimplifiableReservedReg(PhysReg))
-        LiveRegs.addReg(PhysReg);
+    auto Scheduler = DAG->getBB()
+                         ? static_cast<AIEScheduleDAGMI *>(DAG)->getSchedImpl()
+                         : nullptr;
+    const BlockState *BS;
+    if (Scheduler) {
+      MachineBasicBlock *MBB = DAG->getBB();
+      BS = &(Scheduler->getInterBlock().getBlockState(MBB));
+      auto Region = BS->getCurrentRegion();
+      auto BottomRegion = BS->getBottom();
+      if (*Region.begin() == *BottomRegion.begin()) {
+        // If the region is bottom region, liveouts of region are same as
+        // liveouts of MBB
+        for (MCPhysReg Reg : BS->LiveOuts) {
+          LiveRegs.addReg(Reg);
+        }
+      } else {
+        // Else, liveins of succesor region will be the liveouts of current
+        // region
```

Comment:
Not sure it's that easy, there can be a call in between the two regions which clobbers most of the registers. Only the liveins that are also CSRs will remain liveins. However, for our reserved registers, I'm not too sure about their scope, and we should probably assume they are made live by any function call.

For now, I'd suggest to only update the bottom region. If that proves to be insufficient, we can have a follow-up commit that accumulates the liveins of all the regions if there is no function call in between.